src/hipscat_import/cross_match/macauff_arguments.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from os import path
+from typing import List
+
+from hipscat.io import FilePointer, file_io
+from hipscat.io.validation import is_valid_catalog
+
+from hipscat_import.catalog.resume_plan import ResumePlan
+from hipscat_import.runtime_arguments import RuntimeArguments
+
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=unsupported-binary-operation
+
+
+@dataclass
+class MacauffArguments(RuntimeArguments):
+    """Data class for holding cross-match association arguments"""
+    ## Input - Cross-match data
+    input_path: FilePointer | None = None
+    """path to search for the input data"""
+    input_format: str = ""
+    """specifier of the input data format. this will be used to find an appropriate
+    InputReader type, and may be used to find input files, via a match like
+    ``<input_path>/*<input_format>`` """
+    input_file_list: List[FilePointer] = field(default_factory=list)
+    """can be used instead of `input_format` to import only specified files"""
+    input_paths: List[FilePointer] = field(default_factory=list)
+    """resolved list of all files that will be used in the importer"""
+    add_hipscat_index: bool = True
+    """add the hipscat spatial index field alongside the data"""
+    highest_healpix_order: int = 10
+    """healpix order to use when mapping. this will
+    not necessarily be the order used in the final catalog, as we may combine
+    pixels that don't meed the threshold"""
+    pixel_threshold: int = 1_000_000
+    """maximum number of rows for a single resulting pixel.
+    we may combine hierarchically until we near the ``pixel_threshold``"""
+    mapping_healpix_order: int = -1
+    """healpix order to use when mapping. will be
+    ``highest_healpix_order`` unless a positive value is provided for
+    ``constant_healpix_order``"""
+
+    ## Input - Left catalog
+    left_catalog_dir: str = ""
+    left_id_column: str = ""
+    left_ra_column: str = ""
+    left_dec_column: str = ""
+
+    ## Input - Right catalog
+    right_catalog_dir: str = ""
+    right_id_column: str = ""
+    right_ra_column: str = ""
+    right_dec_column: str = ""
+
+    ## `macauff` specific attributes
+    metadata_file_path: str = ""
+    match_probability_column: str = "match_p"
```

Comment:
I think this will be a list of columns - I think there are a few different strategies for computing the match likelihood (spatial, color match, combined).